optimade/validator/utils.py

```diff
@@ -244,16 +256,16 @@ def get(self, request: str):
                     return self.response
 
             # If the connection times out, retry but cache the error
-            except requests.exceptions.ConnectionError as exc:
+            except httpx.ConnectError as exc:
                 errors.append(str(exc))
 
             # Read timeouts should prevent further retries
-            except requests.exceptions.ReadTimeout as exc:
+            except httpx.ReadTimeout as exc:
                 raise ResponseError(str(exc)) from exc
 
-            except requests.exceptions.MissingSchema:
+            except httpx.UnsupportedProtocol as exc:
                 sys.exit(
-                    f"Unable to make request on {self.last_request}, did you mean http://{self.last_request}?"
+                    f"Unable to make request on {self.last_request!r}, did you mean 'https://{self.last_request}'?\nFull error: {exc}"
```

Comment:
When running the validator, should we have an outer `try/except` (perhaps for a custom catch-'em-all `ValidatorException` or similar?) to always add this extra `Full error` part instead? Then we can just raise with an appropriate error message here instead.

Reply:
We already have this with `InternalError` being "anything unexpected", but this is the one case where we want to end all further requests as the URL will remain malformed for the duration. This is unrelated to this PR really